### PR TITLE
Qualification tool: Add regex and username filters.

### PR DESCRIPTION
**Pull request by nartal1**
This PR addresses these feature requests:
1. As a user, I want to select log files for applications whose application name matches a given regular expression so that I can only qualify a subset of applications of interest
2. As a user, I want to select log files for applications submitted by a specific user so that I can qualify only applications a particular user has submitted

This PR fixes https://github.com/NVIDIA/spark-rapids/issues/3041

### CODE REVIEW

**Comment by nartal1** [comment]
build

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualificationArgs.scala [hunk 1]
```diff
@@ -64,9 +64,10 @@ Usage: java -cp rapids-4-spark-tools_2.12-<version>.jar:$SPARK_HOME/jars/*
   val applicationName: ScallopOption[String] =
     opt[String](required = false,
       descr = "Filter event logs whose application name matches " +
-          "exactly or is a substring of input string. Regular expressions not supported." +
-          "For filtering based on complement of application name, use ~APPLICATION_NAME. i.e " +
-          "Select all event logs except the ones which have application name as the input string.")
+          "exactly or is a substring of input string. Application name as regular expression " +
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualificationArgs.scala:67]: perhaps reword this to say something like:
Filter event logs by application name. The string specified can be a regular expression, substring, or exact match....

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/AppFilterSuite.scala [hunk 2]
```diff
@@ -441,4 +441,77 @@ class AppFilterSuite extends FunSuite {
...
+    testAppNameRegexAndUserName(appsWithAppNameRegexAndUserNameToTest,
+      "10-newest", "[Nn].*", "user3", 2)
+  }
+
+  test("App Name Regex exclude with user name match") {
+    testAppNameRegexAndUserName(appsWithAppNameRegexAndUserNameToTest,
+      "10-newest", "[^Nn].*", "user3", 0)
+  }
+
+  test("App Name partial with username match") {
```
[the remark this answers is not in the document]
Comment (nartal1) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/AppFilterSuite.scala:476]: Added test to filter only on username. Please take a look if it's fine or do you want me to add another def `TestUserNameOnly`.

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualificationArgs.scala [hunk 3]
```diff
@@ -64,9 +64,10 @@ Usage: java -cp rapids-4-spark-tools_2.12-<version>.jar:$SPARK_HOME/jars/*
   val applicationName: ScallopOption[String] =
     opt[String](required = false,
       descr = "Filter event logs whose application name matches " +
-          "exactly or is a substring of input string. Regular expressions not supported." +
-          "For filtering based on complement of application name, use ~APPLICATION_NAME. i.e " +
-          "Select all event logs except the ones which have application name as the input string.")
```
[the remark this answers is not in the document]
Comment (nartal1) [reply tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualificationArgs.scala:69]: Thanks! Updated the docs. I had initally updated it but lost it while upmerging.

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualificationArgs.scala (hunk 1 above)
Comment (nartal1) [reply tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/QualificationArgs.scala]: Done

**Comment by tgravescs** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Comment by tgravescs** [comment]
@nartal1 please file issues for these and link them.
